### DISCUSSION TITLE
Problems with implementig livewire in a laravel package

### DISCUSSION THREAD

**Discussion by sharifghaderi** [livewire/livewire#4582] [1. Help] [answered]
Hello everybody.
I have developed some apps with livewire already but now I'm trying to make a package which is requiring livewire and is based on livewire. 

Somehow I can't get it work. 

I have created a StackOverflow question for that which I don't wan't to copy here  again but would be very thankful if somebody could tell me what I'm doing wrong here. 

https://stackoverflow.com/questions/70883108/livewire-looks-for-classes-in-the-root-of-project

**Comment by jonneroelofs** [comment, marked as answer]
The namespace is in your component is not correct. It should be a namespace that points to your package and which you have specified in the composer.json of your package.

Let's say you have this configured in your composer.json:
```json
"autoload": {
    "psr-4": {
      "YourOrganization\\YourPackage\\": "src/"
    }
  },
```

Then in your Sidebar component you would use the namespace:
```php
namespace YourOrganization\YourPackage\Sidebar
```

Also you seem to be using blade component syntax in how you load your livewire component in the ServiceProvider. It should me something like (so no '::' required I think):
```php
Livewire:component('name-you-want-to-load-component-with-in-your-projects',  YourOrganization\YourPackage\Sidebar::class);
```

Because you have this in your service provider:
```php
$this->loadViewsFrom(__DIR__.'/resources/views', 'newpackage');
```

That means that you render function for your component should look like this:
```php
public function render(){
    return view('newpackage::sidebar');
}
```

In all the examples above I assumed you have your Sidebar component located in the root your 'src' folder. If you have it in a subfolder then you have to adjust everything accordingly.

This is how I do it. If this does not resolve everything for you, have a look at open source livewire packages.

Hope this helps.
